plugins/woocommerce/src/Admin/API/MarketingCampaigns.php

```diff
@@ -132,6 +132,65 @@ public function get_items( $request ) {
 		return $response;
 	}
 
+	/**
+	 * Get formatted price based on Price type.
+	 *
+	 * This uses plugins/woocommerce/i18n/currency-info.php to get option object based on $price.currency.
+	 *
+	 * Example: when $price->currency is 'USD' and $price->value is '100', it should return '$100.00'.
+	 *
+	 * @param Price $price Price object.
+	 * @return String formatted price.
+	 */
+	private function get_formatted_price( $price ) {
```

Comment:
`wc_price` would use store currency settings by default, and it will return the price wrapped in HTML tags with encoded HTML entities.

This `get_formatted_price` function will format the price based on `$price->currency`, which may be different from the store currency settings, and it will also strip out the HTML tags and decode the HTML entities so that it can be displayed easily in the React front-end UI.

Reply:
I updated the code comment for the function with examples in https://github.com/woocommerce/woocommerce/pull/44917/commits/964f26590f71c494eff2a1ee7d46dda199be4adc.

Reply:
Quick question:

Why $price may use different than store currency?

Other thing is that your search for the decimal numbers is not precise. You are looking for currency in /i18n/locale-info.php but a particular currency may set for multiple locales. For example EUR is present in the table over 30 times. Your search finds the first occurence. And the decimal separator is not tied to currency but to locale.

Reply:
> Why $price may use different than store currency?

Because marketing campaign may be in a different currency. 

For example, I have a store based in Malaysia with MYR currency. I decide to run a marketing campaign in Japan with the cost of 500 JPY.

Also, users might change their store currency down the road, e.g. I migrate from Malaysia to Singapore, and change the store currency to SGD. In that case, the cost of the marketing campaign should still be 500 JPY.

This is the reason why the `Price` class has `$value` and `$currency`. 

> Other thing is that your search for the **decimal numbers** is not precise. You are looking for currency in /i18n/locale-info.php but **a particular currency may set for multiple locales**. For example EUR is present in the table over 30 times. Your search finds the first occurence. 

Actually the locale does not really matter. What I need is to get the relationship between currency code (e.g. "EUR") and their decimal numbers.

Based on my search on the Internet, each currency would have its own decimal numbers; locale does not matter. 

So even though EUR is present in the table over 30 times, all of them have the same `'num_decimals'   => 2`, so it is fine to use the first occurrence.

(Side note: IMO, I think the `'num_decimals'   => 2` should be in currency-info.php, then we wouldn't have this first occurrence issue.)

> And the decimal separator is not tied to currency but to locale.

Yes, that's true, but to make things simple, right now I'm just using the "default" locale in currency-info.php, which I hope can satisfy the majority of the use cases.

Reply:
> > And the decimal separator is not tied to currency but to locale.
> 
> Yes, that's true, but to make things simple, right now I'm just using the "default" locale in currency-info.php, which I hope can satisfy the majority of the use cases.

As it turns out, it isn't that difficult to use user locale. I made the change in https://github.com/woocommerce/woocommerce/pull/44917/commits/aa5679d32d50838acf51ebe17e683b97cfc93b9f.

Reply:
I have updated the PR description to mention how to test with user locale.